pilot/pkg/bootstrap/server.go

```diff
@@ -247,10 +247,17 @@ func (s *Server) initClusterRegistries(args *PilotArgs) (err error) {
 		s.clusterStore, err = clusterregistry.ReadClusters(s.kubeClient,
 			args.Config.ClusterRegistriesConfigmap,
 			args.Config.ClusterRegistriesNamespace)
+		if err != nil {
+			return err
+		}
 		if s.clusterStore != nil {
 			log.Infof("clusters configuration %s", spew.Sdump(s.clusterStore))
 		}
 	}
+	// Starting Secret controller which will watch for Secret Objects and handle
```

Comment:
I would add an 'else' here .

Reply:
Not sure which if you are referring to, please clarify.